src/azure-cli/azure/cli/command_modules/storage/operations/account.py

```diff
@@ -285,6 +295,18 @@ def update_storage_account(cmd, instance, sku=None, tags=None, custom_domain=Non
             raise CLIError('incorrect usage: --default-action ACTION [--bypass SERVICE ...]')
         params.network_rule_set = acl
 
+    if hasattr(params, 'routing_preference') and any([routing_choice, publish_microsoft_endpoints,
+                                                      publish_internet_endpoints]):
+        if params.routing_preference is None:
+            RoutingPreference = cmd.get_models('RoutingPreference')
+            params.routing_preference = RoutingPreference()
+        if routing_choice is not None:
```

Comment:
>routing_choice [](start = 11, length = 14)

This is just a question, if customer specify --routing-choice without a value, is routing_choice an empty string like "" or none?

Reply:
It will be none.